examples/src/main/scala/zio/config/examples/AutomaticConfigDescExample.scala

```diff
@@ -0,0 +1,54 @@
+package zio.config.examples
+
+import zio.ZIO
+import zio.config.ConfigSource
+import zio.config.magnolia.ConfigDescriptorProvider._
+import zio.console.Console.Live.console._
+
+final case class Token(value: String) extends AnyVal
+final case class Aws(region: String, token: Token)
+final case class MyConfig(
+  aws: Aws,
+  port: Int,
+  dburl: String,
+  port2: Option[Double],
+  price: Either[Double, String],
```

Comment:
It would be good to have not just `Either` here, but a custom sum type, so you can test that sum types are correctly translated into the nested eithers and then iso-mapped to and from the sum type.

Reply:
Yes, I am in the process of doing it.